lgc/patch/PatchLoopMetadata.cpp

```diff
@@ -130,14 +169,49 @@ bool PatchLoopMetadata::runOnModule(Module &module) {
         Metadata *nonforcedMeta[] = {MDString::get(*m_context, "llvm.loop.disable_nonforced")};
         MDNode *nonforcedMetaNode = MDNode::get(*m_context, nonforcedMeta);
         loopMetaNode = MDNode::concatenate(loopMetaNode, MDNode::get(*m_context, nonforcedMetaNode));
+        changed = true;
+      } else if (!StrictUnrollHints) {
+        for (unsigned i = 1, e = loopMetaNode->getNumOperands(); i < e; ++i) {
+          Metadata *op = loopMetaNode->getOperand(i);
+          if (MDNode *mdNode = dyn_cast<MDNode>(op)) {
+            const MDString *mdString = dyn_cast<MDString>(mdNode->getOperand(0));
+            if (mdString)
+              if (mdString->getString().startswith("llvm.loop.unroll.disable")) {
+                LLVM_DEBUG(dbgs() << "  relaxing llvm.loop.unroll.disable\n");
+                // We will use a threshold of 250 to replace the disable hint.
+                int threshold = 250;
+                Metadata *thresholdMeta[] = {
+                    MDString::get(*m_context, "amdgpu.loop.unroll.threshold"),
+                    ConstantAsMetadata::get(ConstantInt::get(Type::getInt32Ty(*m_context), threshold))};
```

Comment:
Most of the code I've seen uses builder to get integer constants

Reply:
There are quite a lot of cases that don't use builder, so I'm not sure I see the advantage here particularly since there is no other use of builder.